Script/pathfinder.lua

```diff
@@ -81,21 +113,25 @@ me.on_script_path_request_finished = function(event)
     local spidertrons = request.spidertrons
     local path = event.path
     if event.try_again_later then
+        -- todo reschedule after n ticks, global.pathfinder_requests should have all required information
         debug_lib.VisualDebugText("pathfinding failed: engine is busy", spidertrons[1])
-    elseif path then
-        local clean_path = me.clean_linear_path(path)
-        for _, spidertron in ipairs(spidertrons) do
-            spidertron.autopilot_destination = nil
-            local x = 0
-            for i, waypoint in ipairs(clean_path) do
-                spidertron.add_autopilot_destination(waypoint.position)
-                debug_lib.VisualDebugCircle(waypoint.position, spidertron.surface, color_lib.color_alpha(color_lib.colors.pink, 0.2), tostring(x))
-                x = x + 1
+        request.try_again_later = true
+    else
+        if path then
+            local clean_path = me.clean_linear_path(path)
+            for _, spidertron in ipairs(spidertrons) do
+                spidertron.autopilot_destination = nil
+                local x = 0
+                for i, waypoint in ipairs(clean_path) do
+                    spidertron.add_autopilot_destination(waypoint.position)
+                    debug_lib.VisualDebugCircle(waypoint.position, spidertron.surface, color_lib.color_alpha(color_lib.colors.pink, 0.2), tostring(x))
+                    x = x + 1
+                end
             end
+        else
+            debug_lib.VisualDebugText("pathfinder callback: path nil", spidertrons[1])
         end
         global.pathfinder_requests[event.id] = nil
```

Comment:
There is a conflict on this one

I put it in the else statement above it because it would then only run if our paths are nil, not every mod that requests a path

Reply:
I put it at that place to not-nil requests that have try-again-later, because we want to re-run them after a few ticks